### PR TITLE
✨(frontend) remove member modal

### PR DESCRIPTION
**Pull request by AntoLC**
## Purpose

Remove a member from a team.

## Proposal

- [x] add endpoint to remove member
- [x] integrate modal, controls and interactions
- [x] tests

## Demo

[scrnli_3_25_2024_5-54-43 PM.webm](https://github.com/numerique-gouv/people/assets/25994652/a5301ad3-b733-4a61-a966-45835964f891)

### CODE REVIEW

**Review by lebaudantoine — COMMENTED** [review]
I mainly focused on the functionalities. I test everything LGTM. However, the translations were not pushed?

File: src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+import { addNewMember, createTeam, keyCloakSignIn } from './common';
+
+test.beforeEach(async ({ page, browserName }) => {
+  await page.goto('/');
+  await keyCloakSignIn(page, browserName);
+});
+
+test.describe('Members Delete', () => {
+  test('it deletes himself when it is the last owner', async ({
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts:11]: `it cannot delete...`

File: src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    await page.getByRole('button', { name: 'Validate' }).click();
+    await expect(
+      page.getByText(`The member has been removed from the team`),
+    ).toBeVisible();
+    await expect(
+      page.getByRole('button', { name: `Create a new team` }),
+    ).toBeVisible();
+  });
+
+  test('it deletes owner member', async ({ page, browserName }) => {
```
Comment (lebaudantoine) [inline src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts:61]: `it cannot delete another owner member`

File: src/frontend/apps/desk/src/features/members/api/useRemoveTeamAccess.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { KEY_LIST_TEAM, KEY_TEAM } from '@/features/teams/';
+
+import { KEY_LIST_TEAM_ACCESSES } from './useTeamsAccesses';
+
+interface RemoveTeamAccessProps {
+  teamId: string;
+  accessId: string;
+}
+
+export const removeTeamAccess = async ({
```
[the remark this answers is not in the document]
Comment (AntoLC) [reply src/frontend/apps/desk/src/features/members/api/useRemoveTeamAccess.ts]: [0cb813c](https://github.com/numerique-gouv/people/pull/147/commits/0cb813cf7917ebb1a49ca182dadfcceb895618e4)

I renamed the commit as well.

File: src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts (hunk 2 above)
Comment (AntoLC) [reply src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts]: https://github.com/numerique-gouv/people/pull/147/commits/f4583b546c820e8cd6c5ca74732c6b7c46dee59a

File: src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts (hunk 1 above)
Comment (AntoLC) [reply src/frontend/apps/e2e/__tests__/app-desk/member-delete.spec.ts]: https://github.com/numerique-gouv/people/pull/147/commits/f4583b546c820e8cd6c5ca74732c6b7c46dee59a

**Review by lebaudantoine — APPROVED** [review]

**Pull request merged**